packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -665,6 +665,12 @@ function updateOffscreenComponent(
   const prevState: OffscreenState | null =
     current !== null ? current.memoizedState : null;
 
+  // Offscreen stores pending changes to visibility in `_pendingVisibility`. This is
+  // to support batching of `attach` and `detach` calls.
+  workInProgress.stateNode._visibility &= ~OffscreenDetached;
+  workInProgress.stateNode._visibility |=
```

Comment:
This should happen in `commitWork`, not `beginWork`.

Reply:
That wouldn't work, would it?

The visibility flag needs to be set before it is read for the first time. The visibility flag is read in begin phase, if it is set in commit work, that's too late. 
If we set it in commit phase, it would take effect on next update.

Reply:
Resolved in https://github.com/facebook/react/pull/25603/commits/bdf8abcbd8c59b610a5617f06dd12f55b0e509dc